docs/guides/1239-export-package-to-repository.md

```diff
@@ -0,0 +1,136 @@
+
+### Practice
+
+Let's try with an archived repository from dagger, in a more down-to-earth manner:
+
+- Initialize the workdir environment
+
+```console
+mkdir packageTest && cd packageTest
+```
+
+- Initialize the project
+
+```console
+dagger project init
+```
+
+- Download the files from archived `examples` repo (just for demo purposes)
+
+```console
+$ dagger project update https://github.com/dagger/examples@main
+5:15PM INF system | installing specified packages...
+5:15PM INF system | installed/updated package github.com/dagger/examples@main
+```
+
+Be careful, we download it just to analyze the behavior. You shouldn't use it, as it has been archived and doesn't contain up-to-date Dagger files.
+
+The files should have been downloaded as shown below:
+
+```console
+tree  -L 6 .
+.
+└── cue.mod
+    ├── dagger.mod
+    ├── dagger.sum
+    ├── module.cue
+    ├── pkg
+    │   └── github.com
+    │       └── dagger
+    │           └── examples
+    │               ├── LICENSE
+    │               ├── Makefile
+    │               ├── README.md
+    │               ├── helloapp
+    │               └── todoapp
+    ├── tmp
+    │   └── github.com
+    │       └── dagger
+    └── usr
+
+11 directories, 6 files
+```
+
+[As you can verify yourself](https://github.com/dagger/examples), the package manager basically downloads what is in the repo and copies it to the proper location.
+
+### Exposing our package
+
+Starting from the previously built package, you should already have below subfolder structure:
+
+```console
+$ tree  -L 6 .
+.
+├── bar
+│   └── anything.cue
+├── cue.mod
+│   ├── module.cue
+│   ├── pkg
+│   └── usr
+│       └── github.com
+│           └── user
+│               └── myrepo
+│                   ├── bar -> /tmp/daggerTest/bar
+│                   └── foo -> /tmp/daggerTest/foo
+├── foo
+│   └── main.cue
+└── main.cue
+
+10 directories, 4 files
+```
+
+As said on the `split plan with package` doc, the `usr` subfolder structure shall follow the url on which you want to push the package.
+
+Why ? Because it will make sure that all of the internal packages referencing each other don't break when being retrieved from a repository.
+
+Let's think about it for a second: if a user was to retrieve both of these packages (`foo` and `bar`) from the package manager, it would copy all of the files shown in the tree above to the `cue.mod/pkg/url` folder.
+
+As, at its root, both `foo/main.cue` and `bar/anything.cue` files were retrieved from the package manager, if the cross-referenced imports in those files don't match the constructed folder structure in `cue.mod/pkg/url`, it would ultimately lead to an import error.
+
+There are two ways to split our current implementation:
+
+1. Lean version: remove everything apart the `foo` and `bar` folders and push them
+2. Full version: push everything as is
+3. Hybrid version: push everything but trim down the number of files using releases
```

Comment:
If you follow what I said above, this step should be much simpler. Just push your already created git repo to a remote. That's because you've already been deciding during development what'll eventually get published (by placing files in which dir).

By this point you should already understand the difference between a dagger project (with a plan that imports packages), and a dagger package (that can be installed with `project update` in a dagger project).